src/classes/NPSP_Address.cls

```diff
@@ -0,0 +1,377 @@
+        return this.addr;
+    }
+
+    public String householdId() {
+        return this.addr.Household_Account__c;
+    }
+
+    public Boolean overlapsWith(NPSP_Address other) {
+        if (other == null) {
+            return false;
+        }
+        if (!isSeasonal() || !other.isSeasonal()) {
+            return false;
+        }
+
+        // we don't need to know what year or month we are currently in, but
+        // we need to take into account days (since we could be in the same month).
+        // to make this easy to compare, we convert month & day to be a decimal of month.day.
+        // eg., 3.01 for Feb 1st and 3.28 for Feb 28th.
+        Decimal start1 = decimalOfMonthDay(startMonth(), startDay());
+        Decimal end1 = decimalOfMonthDay(endMonth(), endDay());
+        Decimal start2 = decimalOfMonthDay(other.startMonth(), other.startDay());
+        Decimal end2 = decimalOfMonthDay(other.endMonth(), other.endDay());
+
+        // now we see if the ranges are increasing (Jan '14 - March '14), or decreasing (Nov 14' - Feb '15),
+        Boolean range1Increases = start1 < end1;
+        Boolean range2Increases = start2 < end2;
+
+        // we can now tell if there is overlap simply by looking at three scenarios:
+        // 1. both ranges increase, make sure start1 not in range2, and start2 not in range1.
+        if (range1Increases && range2Increases) {
+            return (start1 >= start2 && start1 <= end2) || (start2 >= start1 && start2 <= end1);
+        }
+        // 2. one range increases, one range decreases, make sure end1 not in range2, and end2 not in range1.
+        else if (range1Increases != range2Increases) {
+            return (end1 >= start2 && end1 <= end2) || (end2 >= start1 && end2 <= end1);
+        }
+        // 3. both ranges decrease, always implies an overlap of the new year!
+        else {
+            return true;
+        }
+
+    }
+
+    public String endDay() {
+        return addr.Seasonal_End_Day__c;
+    }
+
+    public String endMonth() {
+        return addr.Seasonal_End_Month__c;
+    }
+
+    public String startDay() {
+        return addr.Seasonal_Start_Day__c;
+    }
+
+    public String startMonth() {
+        return addr.Seasonal_Start_Month__c;
+    }
+
+    /*******************************************************************************************************
+    * @description Utility to copy Address fields from an Address object to a Contact or Account.
+    * Handles instances where State and Country picklists are enabled, and multiline street addresses.
+    * @param addr the Address object to copy from
+    * @param sobjDst the destination Contact or Account
+    * @param strFieldPrefix the address fields to copy to, ie., Mailing, Other, Shipping, Billing
+    * @param strFieldAddrType an optional Address Type field on sobjDst to copy to
+    * @return void
+    */
+    public void copyOntoSObject(SObject sobjDst, String strFieldPrefix, String strFieldAddrType) {
```

Comment:
Would this later be replaced in the various entity classes (`NPSP_Account`, `NPSP_Contact`, etc) where we'd initiate and pass in an `NPSP_Address` to copy an address?

Reply:
Yep, exactly! I just pushed a few commits that get us a little closer to that.